# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/sharedworkers/SharedWorkerOp.cpp
# Commit: f2debe329bca
# Full Hash: f2debe329bcaaad1248a234c165ccf240d602344
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2025-04-15 21:21:06
# Regressor Bug: 1899503
# File Overlap Count: 4
# Description:
#   Bug 1899503 - P3 PRemoteWorkerDebugger binding implementation. r=asuth
#   
#   This patch implements the PRemoteWorkerDebugger binding.
#   
#   We are trying to keep the same logic debugger registration/unregistration to minimize the difference between the local debugger(the debugger on the content process main thread)
# ==============================================================================

diff -r 604c4c52574a -r f2debe329bca dom/workers/sharedworkers/SharedWorkerOp.cpp
--- a/dom/workers/sharedworkers/SharedWorkerOp.cpp	Tue Apr 15 09:47:39 2025 +0000
+++ b/dom/workers/sharedworkers/SharedWorkerOp.cpp	Tue Apr 15 09:47:39 2025 +0000
@@ -82,6 +82,10 @@
 
   MOZ_ASSERT(aState.is<Running>() || IsTerminationOp());
 
+  if (mOpArgs.type() == SharedWorkerOpArgs::TSharedWorkerThawOpArgs) {
+    aOwner->SetIsThawing(true);
+  }
+
   RefPtr<SharedWorkerOp> self = this;
   RefPtr<RemoteWorkerChild> owner = aOwner;
 
@@ -97,6 +101,11 @@
         }
 
         self->StartOnMainThread(owner);
+        if (self->mOpArgs.type() ==
+            SharedWorkerOpArgs::TSharedWorkerThawOpArgs) {
+          owner->SetIsThawing(false);
+          owner->RunAllPendingOpsOnMainThread();
+        }
       });
 
   MOZ_ALWAYS_SUCCEEDS(SchedulerGroup::Dispatch(r.forget()));
@@ -151,6 +160,12 @@
   } else {
     MOZ_CRASH("Unknown SharedWorkerOpArgs type!");
   }
+
+#ifdef DEBUG
+  if (!mStarted) {
+    mStarted = true;
+  }
+#endif
 }
 
 void SharedWorkerOp::Start(RemoteWorkerNonLifeCycleOpControllerChild* aOwner,
